File: ED_RRiveraG1_A06/Swift/ED_RRiveraG1_A06/ED_RRiveraG1_A06/node.swift
//
//  node.swift
//  ED_RRiveraG1_A06
//
//  Created by Roberto Rivera on 9/7/17.
//  Copyright © 2017 Riverto. All rights reserved.
//

import Foundation

class node {
    var data:Int = 0
    var next:node?
    init(){
        self.data=0
        self.next=nil
    }
    init(data: Int){
        self.data = data
        self.next=nil
    }
    init(data: Int, next: node){
        self.data = data
        self.next = next
    }
    func toString()->String{
        return String(self.data)
    }
}

File: ED_RRiveraG1_A06/C++/cmake-build-debug/Node.cpp
//
// Created by Roberto Rivera on 9/7/17.
//

#include "Node.h"

Node::Node() {
    this->data = 0;
    this->next = nullptr;
}

Node::Node(int data) {
    this->data=data;
    this->next= nullptr;
}

Node::Node(int data, Node& next) {
    this->data=data;
    this->next=next;
}
